numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,133 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def sliding_window_view(x, shape, axis=None, subok=False, writeable=False):
```

Comment:
I think we should make these kwarg-only, either as
```suggestion
def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
```
or
```suggestion
def sliding_window_view(x, shape, *. axis=None, subok=False, writeable=False):
```
depending on whether we want to force the caller to write the more explicit `sliding_window_view(x, s, axis=a)` instead of `sliding_window_view(x, s, a)`.

Reply:
Good idea! I like more the former since I expect that axis will almost always be provided.